pkg/devfile/detect.go

```diff
@@ -147,11 +148,11 @@ func search(log logr.Logger, a Alizer, localpath string, devfileRegistryURL stri
 		err = &NoDevfileFound{Location: localpath}
 	}
 
-	return devfileMapFromRepo, devfilesURLMapFromRepo, dockerfileContextMapFromRepo, err
+	return devfileMapFromRepo, devfilesURLMapFromRepo, dockerfileContextMapFromRepo, componentPortsMapFromRepo, err
 }
 
 // AnalyzePath checks if a devfile or a dockerfile can be found in the localpath for the given context, this is a helper func used by the CDQ controller
-func AnalyzePath(a Alizer, localpath, context, devfileRegistryURL string, devfileMapFromRepo map[string][]byte, devfilesURLMapFromRepo, dockerfileContextMapFromRepo map[string]string, isDevfilePresent, isDockerfilePresent bool) error {
+func AnalyzePath(a Alizer, localpath, context, devfileRegistryURL string, devfileMapFromRepo map[string][]byte, devfilesURLMapFromRepo, dockerfileContextMapFromRepo map[string]string, componentPortsMapFromRepo map[string][]int, isDevfilePresent, isDockerfilePresent bool) error {
```

Comment:
Can we add a comment to explain what the return values are?

Reply:
Updated.